llvm-external-projects/iree-dialects/lib/Dialect/VectorExt/IR/VectorExtOps.cpp

```diff
@@ -58,6 +58,7 @@ void LayoutIterator::maybeFreezeAndConcatenate(
     if (!state.contains(frozenDim)) {
       frozenDimensions.insert(frozenDim);
       state[frozenDim] = frozenIt;
+      state.ranges[frozenDim] = frozenState.ranges.lookup(frozenDim);
```

Comment:
Thinking about this a little bit more, why do we need to keep track of the range? If a particular dimension is frozen, then we never iterate along that range right?